devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,532 @@
+
+        }
+    }
+	result << createEvent(eventMap)
+	//log.debug "Description ${description} parsed to ${result}"
+	result
+}
+
+private sendEventToChild(childNumber, event) {
+	def child = childDevices?.find { getChildId(it.deviceNetworkId) == childNumber }
+
+	if (child) {
+		log.debug "Sending ${event.name} event to $child.displayName"
+		child?.sendEvent(event)
+	} else {
+		log.debug "Child device $childNumber not found!"
+	}
+}
+
+def setCoolingSetpoint(degrees) {
+	setSetpoint(degrees, COOLING_SETPOINT)
+}
+
+def setHeatingSetpoint(degrees) {
+	setSetpoint(degrees, HEATING_SETPOINT)
+}
+
+def setChildCoolingSetpoint(deviceNetworkId, degrees) {
+	log.debug "deviceNetworkId: ${deviceNetworkId} degrees: ${degrees}"
+	def switchId = getChildId(deviceNetworkId)
+	if (switchId != null) {
+		setSetpoint(degrees, COOLING_SETPOINT_UNOCCUPIED)
+	}
+}
+
+def setChildHeatingSetpoint(deviceNetworkId, degrees) {
+	log.debug "deviceNetworkId: ${deviceNetworkId} degrees: ${degrees}"
+
+	def switchId = getChildId(deviceNetworkId)
+	if (switchId != null) {
+		setSetpoint(degrees, HEATING_SETPOINT_UNOCCUPIED)
+	}
+}
+
+def getChildId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	(split.length > 1) ? split[1] as Integer : null
+}
+
+def setSetpoint(degrees, setpointAttr) {
+	log.debug "degrees: ${degrees}, setpointAttr: ${setpointAttr}"
+	if (degrees != null && setpointAttr != null) {
+		log.debug "temperatureScale: ${temperatureScale}"
+		def celsius = (temperatureScale == "C") ? degrees : fahrenheitToCelsius(degrees)
+		celsius = (celsius as Double).round(2)
+
+		delayBetween([
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, setpointAttr, DataType.INT16, zigbee.convertToHexString(celsius * 100)),
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, setpointAttr)
+		], 500)
+	}
+}
+
+def setThermostatFanMode(mode) {
+	if (state.supportedFanModes?.contains(mode)) {
+		if (isViconicsVT8350() || isSchneiderSE8350()) {
+			switch (mode) {
+				case "on":
+					setFanSpeed(1)
+					break
+				case "auto":
+					setFanSpeed(4)
+					break
+			}
+		} else if (isViconicsVT8650() || isSchneiderSE8650()) {
+			getThermostatFanModeCommands(THERMOSTAT_FAN_MODE_ATTRIBUTE_ID_MAP[mode])
+		}
+	} else {
+		log.debug "Unsupported fan mode $mode"
+	}
+}
+
+def fanOn() {
+	getThermostatFanModeCommands(CUSTOM_FAN_MODE_ON)
+}
+
+def fanAuto() {
+	getThermostatFanModeCommands(CUSTOM_FAN_MODE_AUTO)
+}
+
+def fanCirculate() {
+	getThermostatFanModeCommands(CUSTOM_FAN_MODE_CIRCULATE)
+}
+
+def getThermostatFanModeCommands(mode) {
+	if (mode) {
+		delayBetween([
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE, DataType.ENUM8, mode),
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE)
+		], 500)
+	}
+}
+
+def setFanSpeed(speed) {
+	log.debug "setFanSpeed: ${speed}"
+
+	if (speed == 0 || speed >= 4) { //if by any chance user selects 0 or a value higher than 3, it fan will be set to AUTO
```

Comment:
`it fan will be set to AUTO` -> `the fan will be set to AUTO`

So there is no `off` then?

Reply:
No, the only possible options are: **L**ow, **M**ed, **H**igh, **A**uto, **O**n (user can also configure the device so the concrete "fan sequence" can be used:  On-Auto, L-M-H, L-H, L-M-H-A and L-H-A).